### ISSUE TITLE
Implement VHbb analyzer

### ISSUE THREAD

**Issue by arizzi** [vhbb/nanoAOD-tools#2]
Can be mostly copied from existing VHbbProducer in Heppy

Should compute
- Vtype
- V kinematics
- H jets pairs in different stregies (Dijet, CSV)
- H kinematics with different strategies (needed here? or anyhow for system variations we do it later in the fwk)

anyhting else? please expand below

@perrozzi @scooperstein @capalmer85

**Comment by scooperstein** [comment]
Andrea you mentioned before that the btag re-weighting and JEC's we essentially get for free in the sense that there are already modules implemented that we can borrow. What about the jet energy regression? In 2016 both VHbb frameworks were set up to re-calculate the jet energy regression on the fly, so it is not essential to do it at this level, but would be useful to understand how easily it could be incorporated.

**Comment by capalmer85** [comment]
The H kinematics are probably not needed.  I'm not sure the V kinematics are either.  We compute those in AnalysisTools.  @perrozzi does Xbb use them?

**Comment by arizzi** [comment]
jet energy regression is already computed in the NanoAOD because not all inputs needed will be available.
Concerning how to propagate JER/JES  to the jet energy regression (and possibly other quantities) we (as in "nanoaod developers") would like to implement an event weighting technique so that we avoid having to propagate JER/JES.
Lorenzo Bianchini tested this in ttH (to avoid propagation in MEM ) and found some issue with mass shapes so we need to investigate a bit the feasibility (we may need to bin in "deltaR to closest jet")

In any case I remember that also the technique we used to propagate the JER/JES without recomputing the regression was working ~ok.

**Comment by capalmer85** [comment]
When we have an updated regression we have to start from mini-AOD?

**Comment by arizzi** [comment]
yes, we update it less than once per year so I do not consider this a big
deal (and we computed it on 500K ttbar so we do not need 10B events for
training)

**Comment by capalmer85** [comment]
yes. should be ok.  but I guess we should tell Caterina that she can't use these tnuples from nanaAOD for the 2017 regression.  Maybe she realizes already?  do you know?

**Comment by arizzi** [comment]
well for training we can still make 1M events with nanoAOD++ (i.e. adding what is needed)..

In fact, cross checking I believe almost all inputs are there because we do store SV and lepton PTRel.
The only missing thing is Jet_leadTrackPt (we should check how relevant this variable is for the regression)
